modules/navier_stokes/src/userobjects/HLLCUserObject.C

```diff
@@ -0,0 +1,147 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HLLCUserObject.h"
+#include "NS.h"
+#include "SinglePhaseFluidProperties.h"
+
+namespace nms = NS;
+using MetaPhysicL::raw_value;
+
+registerMooseObject("NavierStokesApp", HLLCUserObject);
+
+InputParameters
+HLLCUserObject::validParams()
+{
+  InputParameters params = InternalSideUserObject::validParams();
+  params.addRequiredParam<UserObjectName>(nms::fluid, "Fluid userobject");
+  params.addClassDescription(
+      "Computes free-flow wave speeds on internal sides, useful in HLLC contexts");
+  return params;
+}
+
+HLLCUserObject::HLLCUserObject(const InputParameters & parameters)
```

Comment:
this wont have the right wave speeds for PCNSFV. Maybe just add porosity as an optional param so we can keep a single object?

Reply:
or text in the docs excluding CNSFV, your call

Reply:
I elected for the former